kani-driver/src/call_cargo.rs

```diff
@@ -196,9 +190,10 @@ impl KaniSession {
                             }
                         }
                     },
-                    Message::CompilerArtifact(_)
-                    | Message::BuildScriptExecuted(_)
-                    | Message::BuildFinished(_) => {
+                    Message::CompilerArtifact(rustc_artifact) => {
+                        artifact = Some(rustc_artifact);
```

Comment:
Is it guaranteed that there's a single artifact per cargo invocation? If so, I suggest adding an assertion that checks that, e.g:
```rust
assert_eq!(artifact, None);
```

Reply:
Not really. Every package will generate at least one artifact. However, we only care about the last one, which is the one that we pass the codegen flag to. Thus, I am only running the map in the last artifact that was collected.

Reply:
I see. Perhaps add a comment to mention this?

Reply:
Done